clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/security/AzureCredentials.groovy

```diff
@@ -37,50 +39,26 @@ class AzureCredentials {
   final String clientId
   final String appKey
   final String project
+  final String subscriptionId
 
-  final AzureResourceManagerClient resourceManagerClient
-  final AzureNetworkClient networkClient
-  final AzureComputeClient computeClient
+  final AzureResourceManagerClient resourceManagerClient = null
+  final AzureNetworkClient networkClient = null
+  final AzureComputeClient computeClient = null
 
-  AzureCredentials(String tenantId, String clientId, String appKey,
-                   AzureResourceManagerClient resourceManagerClient,
-                   AzureNetworkClient networkClient,
-                   AzureComputeClient computeClient) {
+  AzureCredentials(String tenantId, String clientId, String appKey, String subscriptionId) {
     this.tenantId = tenantId
     this.clientId = clientId
     this.appKey = appKey
+    this.subscriptionId = subscriptionId
     this.project = "AzureProject"
-    this.resourceManagerClient = resourceManagerClient
-    this.networkClient = networkClient
-    this.computeClient = computeClient
 
-  }
+    resourceManagerClient = new AzureResourceManagerClient(this.subscriptionId,
```

Comment:
It's because of the initial null. "finals" can be assigned once and only once. I've removed the initialization to null in the declaration and the warnings have gone away.